src/CodeGen_Hexagon.cpp

```diff
@@ -1297,8 +1312,17 @@ void CodeGen_Hexagon::visit(const Div *op) {
 }
 
 void CodeGen_Hexagon::visit(const Cast *op) {
-    // TODO: Do we need to handle same-sized vector casts before LLVM sees them?
-    CodeGen_Posix::visit(op);
+    if (op->value.type().bits() == 1 && op->type.bits() > 1) {
+        // If left to LLVM, this maps true to one. However,
+        // eliminate_bool_vectors assumes that it maps to negative
+        // one. This will map true to negative one instead.
+        value = call_intrin(op->type,
```

Comment:
It's not what you would get on OpenCL at least:

> For scalar types, the relational operators shall return 0 if the specified relation is false and 1 if the specified
relation is true. For vector types, the relational operators shall return 0 if the specified relation is false and –1 (i.e. all bits set) if the specified relation is true. 

https://www.khronos.org/registry/cl/specs/opencl-2.0-openclc.pdf, Page 28